lib/spack/spack/environment/environment.py

```diff
@@ -1081,43 +1033,74 @@ def add_view(name, values):
         if self.views == dict():
             self.views[default_view_name] = ViewDescriptor(self.path, self.view_path_default)
 
-    def _process_concrete_includes(self):
-        """Extract and load into memory included concrete spec data."""
-        _included_concrete_envs = self.manifest[TOP_LEVEL_KEY].get(included_concrete_name, [])
-        # Expand config and environment variables
-        self.included_concrete_envs = [
-            spack.util.path.canonicalize_path(_env) for _env in _included_concrete_envs
-        ]
-
-        if self.included_concrete_envs:
+    def _load_concrete_include_data(self):
+        if self.included_concrete_env_roots:
             if os.path.exists(self.lock_path):
                 with open(self.lock_path, encoding="utf-8") as f:
                     data = self._read_lockfile(f)
 
-                if included_concrete_name in data:
-                    self.included_concrete_spec_data = data[included_concrete_name]
+                if lockfile_include_key in data:
+                    self.included_concrete_spec_data = data[lockfile_include_key]
             else:
                 self.include_concrete_envs()
 
+    def _process_included_lockfiles(self):
+        """Extract and load into memory included lock file data."""
+        includes = self.manifest[TOP_LEVEL_KEY].get(lockfile_include_key, [])
+        if includes:
+            tty.warn(
+                f"Use of '{lockfile_include_key}' in manifest files "
+                f"is deprecated. Please change the key to '{manifest_include_name}'."
+                f" And make sure to end each path with '{lockfile_name}'."
```

Comment:
needs an update() method in config schema.

Reply:
I've added support and documentation for this. Unfortunately, I also introduced a circular import when I decided it was important to support path variable substitutions for the destination of an include.

Reply:
Added a "workaround" to avoid the new circular import by using `canonicalize_path`, which is what is consistent with the API used against the `repos` destination.